procrastinate/retry.py

```diff
@@ -43,8 +109,34 @@ def get_schedule_in(self, *, exception: BaseException, attempts: int) -> int | N
             If a job should not be retried, this function should return None.
             Otherwise, it should return the duration after which to schedule the
             new job run, *in seconds*.
+
+        Notes
+        -----
+        This function is deprecated and will be removed in a future version. Use
+        `get_retry_decision` instead.
         """
-        raise NotImplementedError()
+        raise NotImplementedError
+
+    def get_retry_decision(
+        self, *, exception: BaseException, job: Job
+    ) -> RetryDecision | None:
+        """
+        Parameters
+        ----------
+        exception:
+            The exception raised by the job
+        job:
+            The current job
+
+        Returns
+        -------
+        ``RetryDecision``
+            A RetryDecision object with the following attributes:
+            - should_retry: a boolean indicating whether the job should be retried
+            - schedule_in: an optional float indicating the number of seconds to wait
+            - priority: an optional integer indicating the new priority of the job
```

Comment:
This is outdated. We should not list the attributes here but rather point to the corresponding doc.

Reply:
I'll just add the `RetryDecision` to the `reference.rst`. From looking at the reference documentation, it seems that when not providing a `Returns` in the docstring, it automatically links to the correct reference section by evaluating the return type hint.

Reply:
Alright, I updated the API reference. Could you have another look at it? By the way, maybe we should add `TimeDeltaParams` directly to the reference. Currently, it is only mentioned as an aside in the [`Task.configure`](https://procrastinate.readthedocs.io/en/stable/reference.html#procrastinate.tasks.Task.configure) method. But I would do this in another PR, if you agree.